diff -ruN X11-driver-firegl-3.12.0./panel_src/main.cpp X11-driver-firegl-3.12.0/panel_src/main.cpp
--- X11-driver-firegl-3.12.0./panel_src/main.cpp	2004-09-23 16:25:46.000000000 +0200
+++ X11-driver-firegl-3.12.0/panel_src/main.cpp	2004-09-23 16:26:12.689848008 +0200
@@ -28,8 +28,6 @@
 //    printf ( "%s\n", lpath.data() ) ;
 //    printf ( "%s\n", lfile.data() ) ;
 
-    QApplication::setStyle ( "Windows" ) ;
-
     QApplication a( argc, argv );
 
     if ( fi.exists() )
