--- xorg-driver-video-radeon-crimson-15.12/common/lib/modules/fglrx/build_mod/firegl_public.c.b	2016-03-30 21:53:04.123817514 +0200
+++ xorg-driver-video-radeon-crimson-15.12/common/lib/modules/fglrx/build_mod/firegl_public.c	2016-03-30 21:54:42.922113915 +0200
@@ -6452,7 +6452,11 @@
       if (!(fpu->state->xsave.xsave_hdr.xstate_bv & XSTATE_FP))
 #else
 	  _copy_xregs_to_kernel(&fpu->state.xsave);
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4,4,0)
       if (!(fpu->state.xsave.header.xfeatures & XSTATE_FP))
+#else
+      if (!(fpu->state.xsave.header.xfeatures & XFEATURE_MASK_FP))
+#endif
 #endif
          return 1;
    } else if (static_cpu_has(X86_FEATURE_FXSR)) {
