--- fglrx-14.501.1003/common/lib/modules/fglrx/build_mod/firegl_public.c.orig	2015-03-30 21:42:57.000000000 +0200
+++ fglrx-14.501.1003/common/lib/modules/fglrx/build_mod/firegl_public.c	2015-03-30 21:47:52.302266695 +0200
@@ -4816,8 +4816,13 @@
 {
     unsigned long orig_level;
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3,19,0)
+    orig_level = __this_cpu_read(kasExecutionLevel);
+    __this_cpu_write(kasExecutionLevel, level);
+#else
     orig_level = __get_cpu_var(kasExecutionLevel);
     __get_cpu_var(kasExecutionLevel) = level;
+#endif
 
     return orig_level;
 }
@@ -4829,7 +4834,11 @@
  */
 static unsigned long kas_GetExecutionLevel(void)
 {
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3,19,0)
+    return __this_cpu_read(kasExecutionLevel);
+#else
     return __get_cpu_var(kasExecutionLevel);
+#endif
 }
 
 /** \brief Type definition for kas_spin_lock() parameter */
