--- xorg-driver-video-fglrx-8.34.8/common/lib/modules/fglrx/build_mod/firegl_public.c~	2007-02-21 19:15:09.217722366 +0100
+++ xorg-driver-video-fglrx-8.34.8/common/lib/modules/fglrx/build_mod/firegl_public.c	2007-02-23 12:37:13.080613213 +0100
@@ -5329,7 +5329,11 @@
     kasThread_t* thread_obj = (kasThread_t*)hThread;
     init_MUTEX(&(thread_obj->sleep_finished));
     init_waitqueue_head(&(thread_obj->wq_head));
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,20)
+    INIT_WORK(&(thread_obj->work), routine);
+#else
     INIT_WORK(&(thread_obj->work), routine, pcontext);
+#endif
     schedule_work(&(thread_obj->work));
     return 1;
 }
