diff -ru --minimal xorg-driver-video-radeon-crimson-15.12.orig/common/usr/share/applications/amdcccle.desktop xorg-driver-video-radeon-crimson-15.12/common/usr/share/applications/amdcccle.desktop
--- xorg-driver-video-radeon-crimson-15.12.orig/common/usr/share/applications/amdcccle.desktop	2015-12-18 19:47:41.000000000 +0100
+++ xorg-driver-video-radeon-crimson-15.12/common/usr/share/applications/amdcccle.desktop	2016-03-30 20:00:13.956159567 +0200
@@ -3,9 +3,8 @@
 Type=Application
 Exec=amdcccle
 Icon=ccc_large
-Categories=GNOME;KDE;GTK;Qt;HardwareSettings;Settings;
-
-
+Terminal=false
+Categories=Qt;HardwareSettings;Settings;
 Name=AMD Control Center
 Comment=Allows the configuration of the AMD graphics hardware settings.
 Name[cs]=AMD Control Center
diff -ru --minimal xorg-driver-video-radeon-crimson-15.12.orig/common/usr/share/applications/amdccclesu.desktop xorg-driver-video-radeon-crimson-15.12/common/usr/share/applications/amdccclesu.desktop
--- xorg-driver-video-radeon-crimson-15.12.orig/common/usr/share/applications/amdccclesu.desktop	2015-12-18 19:47:41.000000000 +0100
+++ xorg-driver-video-radeon-crimson-15.12/common/usr/share/applications/amdccclesu.desktop	2016-03-30 19:59:02.468312469 +0200
@@ -2,9 +2,9 @@
 Encoding=UTF-8
 Type=Application
 Exec=amdxdg-su -c amdcccle
-Categories=GNOME;KDE;GTK;Qt;HardwareSettings;Settings;
+Categories=Qt;HardwareSettings;Settings;
 Icon=ccc_large
-
+Terminal=false
 Name=AMD Control Center (Administrative)
 Comment=Allows the configuration of the AMD graphics hardware settings that requires administrative access.
 Name[cs]=AMD Control Center (správce)
