--- common/lib/modules/fglrx/build_mod/firegl_public.c.orig	2012-01-21 21:44:22.000000000 +0100
+++ common/lib/modules/fglrx/build_mod/firegl_public.c	2012-01-21 22:21:45.126304477 +0100
@@ -5791,16 +5791,7 @@
  */
 void ATI_API_CALL KCL_fpu_begin(void)
 {
-#ifdef CONFIG_X86_64
     kernel_fpu_begin();
-#else
-    struct thread_info *cur_task = current_thread_info();
-    preempt_disable();
-    if (cur_task->status & TS_USEDFPU)
-        __save_init_fpu(cur_task->task);
-    else
-        clts();
-#endif
 }
 
 /** \brief End of using FPU
